rstudio/rstudio-entrypoint.sh

```diff
@@ -21,9 +21,14 @@ if [ $(find /workspace -type f -name "*.Rproj" | wc -w) -eq 1 ]; then
   cat /home/rstudio/rstudio-rprofile.R >> /home/rstudio/.Rprofile
 fi
 
-# Set file line endings as crlf if docker run from Windows
+# Set RStudio session user settings
+#  - Enable autosave on blur - auto save when editor loses focus
+#  - Enable autosave on idle - commit changes to documents on idle
+#  - Set file line endings as crlf if docker run from Windows
 if [ "$HOSTPLATFORM" = "win32" ] || [ "$HOSTPLATFORM" = "windows" ]; then
-  echo -e "{\n\t\"line_ending_conversion\": \"windows\"\n}" >> /etc/rstudio/rstudio-prefs.json
+  echo -e "{\n\t\"line_ending_conversion\": \"windows\",\n\t\"auto_save_on_blur\": true,\n\t\"auto_save_on_idle\": \"commit\"\n}" >> /etc/rstudio/rstudio-prefs.json
```

Comment:
Ugh, the windows thing complicates stuff doesn't it?

I don't think we should use `>>` here, though? Probably should be `>`, to ensure a valid json document?

Reply:
Yes, I think if we had the `jq` command within the image we could have used that to insert key-value pairs into the json file in a nicer way - but simple enough just to write the whole file with `echo` atm.

Great spot - sorry about that - I'll amend `>>` to `>`.